src/Sylius/Behat/Context/Ui/Admin/ManagingProductVariantsContext.php

```diff
@@ -0,0 +1,312 @@
+use Sylius\Behat\Service\NotificationCheckerInterface;
+use Sylius\Behat\Service\Resolver\CurrentPageResolverInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ */
+final class ManagingProductVariantsContext implements Context
+{
+    /**
+     * @var SharedStorageInterface
+     */
+    private $sharedStorage;
+
+    /**
+     * @var CreatePageInterface
+     */
+    private $createPage;
+
+    /**
+     * @var IndexPageInterface
+     */
+    private $indexPage;
+
+    /**
+     * @var UpdatePageInterface
+     */
+    private $updatePage;
+
+    /**
+     * @var CurrentPageResolverInterface
+     */
+    private $currentPageResolver;
+
+    /**
+     * @var NotificationCheckerInterface
+     */
+    private $notificationChecker;
+
+    /**
+     * @param SharedStorageInterface $sharedStorage
+     * @param CreatePageInterface $createPage
+     * @param IndexPageInterface $indexPage
+     * @param UpdatePageInterface $updatePage
+     * @param CurrentPageResolverInterface $currentPageResolver
+     * @param NotificationCheckerInterface $notificationChecker
+     */
+    public function __construct(
+        SharedStorageInterface $sharedStorage,
+        CreatePageInterface $createPage,
+        IndexPageInterface $indexPage,
+        UpdatePageInterface $updatePage,
+        CurrentPageResolverInterface $currentPageResolver,
+        NotificationCheckerInterface $notificationChecker
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->createPage = $createPage;
+        $this->indexPage = $indexPage;
+        $this->updatePage = $updatePage;
+        $this->currentPageResolver = $currentPageResolver;
+        $this->notificationChecker = $notificationChecker;
+    }
+
+    /**
+     * @Given /^I want to create a new product variant for (this product)$/
+     */
+    public function iWantToCreateANewProduct(ProductInterface $product)
+    {
+        $this->createPage->open(['productId' => $product->getId()]);
+    }
+
+    /**
+     * @When I specify its code as :code
+     * @When I do not specify its code
+     */
+    public function iSpecifyItsCodeAs($code = null)
+    {
+        $this->createPage->specifyCode($code);
+    }
+
+    /**
+     * @When I name it :name
+     */
+    public function iNameItIn($name)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I rename it to :name
+     */
+    public function iRenameItTo($name)
+    {
+        $this->createPage->nameIt($name);
+    }
+
+    /**
+     * @When I add it
+     * @When I try to add it
+     */
+    public function iAddIt()
+    {
+        $this->createPage->create();
+    }
+
+    /**
+     * @When /^I specify its price to ("(?:€|£|\$)[^"]+")$/
```

Comment:
``specify`` sounds incorrect to me in context of price. I think "I set its price to" sounds more natural.